src/firewall.py

```diff
@@ -0,0 +1,152 @@
+        except ValueError as exc:
+            raise ValueError(f"incorrect firewall entry format: {entry}") from exc
+        return cls(ip_range=entry)
+
+
+class Firewall:  # pylint: disable=too-few-public-methods
+    """Represent a firewall and provides methods to refresh its configuration."""
+
+    _ACL_RULESET_NAME = "github"
+
+    def __init__(self, network: str):
+        """Initialize a new Firewall instance.
+
+        Args:
+            network: The LXD network name.
+        """
+        self._network = network
+
+    def get_host_ip(self) -> str:
+        """Get the host IP address for the corresponding LXD network.
+
+        Returns:
+            The host IP address.
+        """
+        address = execute_command(
+            ["/snap/bin/lxc", "network", "get", self._network, "ipv4.address"]
+        )
+        return str(ipaddress.IPv4Interface(address.strip()).ip)
+
+    def refresh_firewall(self, denylist: typing.List[FirewallEntry]):
+        """Refresh the firewall configuration.
+
+        Args:
+            denylist: The list of FirewallEntry objects to allow.
+        """
+        current_acls = [
+            acl["name"]
+            for acl in yaml.safe_load(
+                execute_command(["lxc", "network", "acl", "list", "-f", "yaml"])
+            )
+        ]
+        if self._ACL_RULESET_NAME not in current_acls:
+            execute_command(["/snap/bin/lxc", "network", "acl", "create", self._ACL_RULESET_NAME])
+        execute_command(
+            [
+                "/snap/bin/lxc",
+                "network",
+                "set",
+                self._network,
+                f"security.acls={self._ACL_RULESET_NAME}",
+            ]
+        )
+        execute_command(
+            [
+                "/snap/bin/lxc",
+                "network",
+                "set",
+                self._network,
+                "security.acls.default.egress.action=allow",
+            ]
+        )
+        acl_config = yaml.safe_load(
+            execute_command(["/snap/bin/lxc", "network", "acl", "show", self._ACL_RULESET_NAME])
+        )
+        host_ip = self.get_host_ip()
+        egress_rules = [
+            {
+                "action": "reject",
+                "destination": host_ip,
+                "destination_port": "1-8079,8081-65535",
+                "protocol": "tcp",
+                "state": "enabled",
+            },
+            {
+                "action": "reject",
+                "destination": host_ip,
+                "protocol": "udp",
+                "state": "enabled",
+            },
+            {
+                "action": "reject",
+                "destination": host_ip,
+                "protocol": "icmp4",
+                "state": "enabled",
+            },
+            {
+                "action": "reject",
+                "destination": "::/0",
+                "state": "enabled",
+            },
+        ]
+        host_network = ipaddress.IPv4Network(host_ip)
```

Comment:
The variable name could be confusing, it's not the real host_network, but the host_ip represented as a /32 CIDR notation to be able to use `address_exclude` that requires an IPv4Network object

Reply:
Yes, the variable is a bad idea, I removed the variable and inlined the IPv4Network casting.